src/amqp.js

```diff
@@ -0,0 +1,344 @@
+      debugDev(`Nack message ${message.properties.correlationId}`);
+      channel.nack(message, false, true);
+    });
+    await setTimeoutPromise(10);
+    return;
+  }
+
+  // MARK: Send to queue
+  /**
+   * Sends to queue
+   * Send to queue uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status, data
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @param {object} params.data Data object sent to queue
+   * @returns {boolean} returns true on success
+   */
+  async function sendToQueue({blobId, status, data}) {
+    debug(`sendToQueue`);
+    // eslint-disable-next-line no-useless-catch
+    try {
+      const {queue} = await generateQueueId({blobId, status});
+      debug(`Queue ${queue}`);
+      debugDev(`Queue Asserted: ${queue}`);
+
+      debugData(`Data ${JSON.stringify(data)}`);
+
+      debugDev(`Actually sendToQueue: ${queue}`);
+      await channel.sendToQueue(
+        queue,
+        Buffer.from(JSON.stringify({data})),
+        {
+          correlationId: blobId,
+          persistent: true
+        }
+      );
+      debugDev(`Send message to queue: ${queue}`);
+
+      return true;
+    } catch (error) {
+      debug(`SendToQueue errored: ${JSON.stringify(error)}`);
+      handleAmqpErrors(error);
+    }
+  }
+
+  async function removeQueue({blobId, status}, allowMessageLoss = false) {
+    const {queue, channelInfo} = await generateQueueId({blobId, status});
+    // deleteQueue borks the channel if the queue does not exist
+    // -> use throwaway tempChannel to avoid killing actual channel in use
+    // this might be doable also with assertQueue before deleteQueue
+    const tempChannel = await connection.createChannel();
+    if (!allowMessageLoss && channelContainsMessages(channelInfo)) {
+      throw new Error('Trying to remove queue that has unhandled messages!');
+    }
+
+    await tempChannel.deleteQueue(queue);
+
+    if (tempChannel) {
+      await tempChannel.close();
+      return;
+    }
+
+    return;
+  }
+
+  // ----------------
+  // Helper functions
+  // ----------------
+
+  function channelContainsMessages(channelInfo) {
+    if (channelInfo.messageCount < 1) {
+      debug(`checkQueue: ${channelInfo.messageCount} - queue is empty`);
+      return false;
+    }
+    debug(`Queue has ${channelInfo.messageCount} messages`);
+    return true;
+  }
+
+  async function generateQueueId({blobId, status}) {
+    const isUndefined = blobId === undefined || status === undefined;
+    const notString = typeof blobId !== 'string' || typeof status !== 'string';
+    const isEmpty = blobId.length < 1 || status.length < 1;
+    if (isUndefined || notString || isEmpty) {
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, 'invalid operation parameters');
+    }
+    const queue = `${blobId}.${status}`;
```

Comment:
In REST-api we have queues named '{status}.{correlationId}' - here we have '{blobId}.{status}' - do we want to use same convertion and change order in either REST-api of MRI?